prbench/src/prbench/envs/geom3d/ground3d.py

```diff
@@ -0,0 +1,247 @@
+"""PyBullet environment where an object must be picked from the ground.
+
+There may be other obstructing objects in the environment.
+"""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Type as TypingType
+
+import numpy as np
+from pybullet_helpers.geometry import Pose, set_pose
+from pybullet_helpers.inverse_kinematics import check_body_collisions
+from pybullet_helpers.utils import create_pybullet_block
+from relational_structs import Object, ObjectCentricState
+from relational_structs.utils import create_state_from_dict
+
+from prbench.core import ConstantObjectPRBenchEnv, FinalConfigMeta
+from prbench.envs.geom3d.base_env import (
+    Geom3DEnvConfig,
+    ObjectCentricGeom3DRobotEnv,
+)
+from prbench.envs.geom3d.object_types import (
+    Geom3DCuboidType,
+    Geom3DEnvTypeFeatures,
+    Geom3DRobotType,
+)
+from prbench.envs.geom3d.utils import Geom3DObjectCentricState
+from prbench.envs.utils import PURPLE
+
+
+@dataclass(frozen=True)
+class Ground3DEnvConfig(Geom3DEnvConfig, metaclass=FinalConfigMeta):
+    """Config for Ground3DEnv()."""
+
+    # World bounds.
+    x_lb: float = -1
+    x_ub: float = 1
+    y_lb: float = -1
+    y_ub: float = 1
+
+    # Blocks.
+    block_size: float = 0.02  # cubes (height = width = length)
+    block_rgba: tuple[float, float, float, float] = PURPLE + (1.0,)
+
+
+class Ground3DObjectCentricState(Geom3DObjectCentricState):
+    """A state in the GroundMotion3DEnv().
+
+    Adds convenience methods on top of Geom3DObjectCentricState().
+    """
+
+    def get_cuboid_half_extents(self, name: str) -> tuple[float, float, float]:
```

Comment:
we should be able to use the `get_object_half_extents` and `get_object_pose` methods that I added to the parent class at the end of our pair programming